jwst/ami/oifits.py

```diff
@@ -41,44 +42,208 @@ def __init__(self, fringefitter, method="median"):
         self.nslices = len(self.fringe_fitter.nrm_list)  # n ints
         self.n_baselines = int(comb(self.n_holes, 2))  # 21
         self.n_closure_phases = int(comb(self.n_holes, 3))  # 35
-        self.n_closure_amplitudes = int(comb(self.n_holes, 4))
+        self.n_closure_amplitudes = int(comb(self.n_holes, 4)) # also 35
 
         self.method = method
 
         self.ctrs_eqt = self.fringe_fitter.instrument_data.ctrs_eqt
         self.ctrs_inst = self.fringe_fitter.instrument_data.ctrs_inst
-        self.pa = (
-            self.fringe_fitter.instrument_data.pav3
-        )  # header pav3, not including v3i_yang??
 
         self.bholes, self.bls = self._makebaselines()
         self.tholes, self.tuv = self._maketriples_all()
+        self.qholes, self.quads = self._makequads_all()
 
     def make_obsarrays(self):
         """
         Make arrays of observables of the correct shape for saving to datamodels
         """
-        # arrays of observables, (nslices,nobservables) shape.
+        # empty arrays of observables, (nslices,nobservables) shape.
         self.fringe_phases = np.zeros((self.nslices, self.n_baselines))
         self.fringe_amplitudes = np.zeros((self.nslices, self.n_baselines))
         self.closure_phases = np.zeros((self.nslices, self.n_closure_phases))
+        self.t3_amplitudes = np.zeros((self.nslices, self.n_closure_phases))
+        self.q4_phases = np.zeros((self.nslices, self.n_closure_amplitudes))
         self.closure_amplitudes = np.zeros((self.nslices, self.n_closure_amplitudes))
         self.pistons = np.zeros((self.nslices, self.n_holes))
         # model parameters
         self.solns = np.zeros((self.nslices, 44))
 
+        # populate with each integration's observables
         for i, nrmslc in enumerate(self.fringe_fitter.nrm_list):
-            self.fringe_phases[i, :] = np.rad2deg(nrmslc.fringephase)  # FPs in degrees
+            self.fringe_phases[i, :] = nrmslc.fringephase  # FPs in radians
             self.fringe_amplitudes[i, :] = nrmslc.fringeamp
-            self.closure_phases[i, :] = np.rad2deg(nrmslc.redundant_cps)  # CPs in degrees
+            self.closure_phases[i, :] = nrmslc.redundant_cps  # CPs in radians
+            self.t3_amplitudes[i, :] = nrmslc.t3_amplitudes
+            self.q4_phases[i, :] = nrmslc.q4_phases # quad phases in radians
             self.closure_amplitudes[i, :] = nrmslc.redundant_cas
-            self.pistons[i, :] = np.rad2deg(
-                nrmslc.fringepistons
-            )  # segment pistons in degrees
+            self.pistons[i, :] = nrmslc.fringepistons # segment pistons in radians
```

Comment:
Could you give an example of how I'd do this? I'm not familiar with np.newaxis but it seems useful and the first way I tried it wasn't right.

Reply:
I think I didn't read this well the first time, I missed the enumerate.  I would leave this as-is.  Sorry about that.
In order to avoid the for loop I think you'd need to do a much more substantial refactor.

Reply:
No worries, I'm glad to hear it wasn't just my inability to figure out how to use np.newaxis properly!